beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -487,13 +491,29 @@ func (f *blocksFetcher) fetchBlobsFromPeer(ctx context.Context, bwb []blocks2.Bl
 	if req == nil {
 		return bwb, nil
 	}
-	// Request blobs from the same peer that gave us the blob batch.
-	blobs, err := f.requestBlobs(ctx, req, pid)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not request blobs by range")
+	peers = f.filterPeers(ctx, peers, peersPercentagePerRequest)
+	// We dial the initial peer first to ensure that we get the desired set of blobs.
+	wantedPeers := append([]peer.ID{pid}, peers...)
+	bestPeers := f.hasSufficientBandwith(wantedPeers, req.Count)
+	// We append the best peers to the front so that higher capacity
+	// peers are dialed first.
+	peers = append(bestPeers, peers...)
```

Comment:
Won't this result in `bestPeers` appearing twice in the slice?

Reply:
The initial idea is to provide peers with throughput and if all of them fail we just fallback to the whole peer set. It does duplicate it, but I don't see too much benefit in removing duplicates

Reply:
If we do decide to request from multiple peers in a loop (despite my other comment), the first call to `requestBlob` for a peer that wasn't in the `hasSufficientBandwidth` result will call `waitForBandwidth`, causing the thread to block. In the worst case this is one of the `bestPeers`, which may have a longer wait because we just made a request to it.

Instead of doing that, I would suggest only appending `pid` - ie if none of the peers that seem to have bandwidth in the first check are able to serve the blobs, wait until the peer that served you the blocks has bandwidth.